core/device/networkstate.go

```diff
@@ -45,14 +45,13 @@ func (n *networkState) GetNetworkState() model.DeviceNetworkType {
 
 func (n *networkState) SetNetworkState(networkState model.DeviceNetworkType) {
 	n.networkMu.Lock()
-	defer n.networkMu.Unlock()
-
 	if n.networkState == networkState {
 		// to avoid unnecessary hook calls
 		return
 	}
 	n.networkState = networkState
-	n.runOnNetworkUpdateHook()
+	n.networkMu.Unlock()
+	n.runOnNetworkUpdateHook(networkState)
```

Comment:
deadlock in case of networkState has not changed

Reply:
I'll remove this code, it was just a test and doesn't relate to progress bar. May be I committed it by mistake